.circleci/config.yml

```diff
@@ -164,15 +164,16 @@ jobs:
     steps:
       - checkout
       - restore_cache:
-          key: python-requirements-{{ checksum "requirements-dev.txt" }}
+          key: python-requirements-{{ checksum "requirements-typehint.txt" }}
       - run:
           name: Setup environment
           command: |
+            make install-dev-types
```

Comment:
@k-a-il @silv-io Now that the CircleCI pipeline is being migrated to GitHub Actions I think this would mean that we also have to change the make target here, right?
https://github.com/localstack/localstack/blob/9b6ef35c76a0685f8061e09f3145141c12368ad6/.github/actions/setup-tests-env/action.yml#L15

Reply:
Yes, would be great to add this in this PR.

On the other hand I think we should also be cautious with this switch because the typehint dependencies take waaaay longer to install in my experience (except when using `uv` :D)

Reply:
Well, is this a blocker then? What does "waaaay longer" mean here?

Reply:
Yes :) 

- I'll add a suggestion in my review
- Then, we can manually run the `AWS / Build, Test, Push` pipeline on this PR to check if it's green. (We've still got some unrelated test failures around the Docker SDK there. These will not be blocking this PR)
- I've checked the CircleCI Pipeline and it looks like it will add 30s of installation time. Seems like caching works a bit better here than on-device, so the impact is not as big. The trade-off is fine in this case.